app/src/main/java/org/fossasia/phimpme/trashbin/TrashBinActivity.java

```diff
@@ -0,0 +1,111 @@
+
+    @BindView(R.id.trashbin_recycler_view)
+    RecyclerView trashBinRecyclerView;
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @BindView(R.id.empty_trash)
+    RelativeLayout emptyView;
+
+    @BindView(R.id.swipeRefreshLayout_trashbin)
+    protected SwipeRefreshLayout swipeRefreshLayout;
+
+    @BindView(R.id.view_parent)
+    RelativeLayout parentView;
+
+    private RealmQuery<TrashBinRealmModel> trashBinRealmModelRealmQuery;
+    private TrashBinAdapter trashBinAdapter;
+
+    @Override public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_trash_bin2);
+        ButterKnife.bind(this);
+        Realm realm = Realm.getDefaultInstance();
+        trashBinRealmModelRealmQuery = realm.where(TrashBinRealmModel.class);
+        ArrayList<TrashBinRealmModel> trashlist = getTrashObjects();
+        if(trashlist.size() == 0){
+            emptyView.setVisibility(View.VISIBLE);
+        }else{
+            trashBinAdapter = new TrashBinAdapter(trashlist);
+            GridLayoutManager layoutManager = new GridLayoutManager(getApplicationContext(), columnsCount());
+            layoutManager.setReverseLayout(false);
+            trashBinRecyclerView.setLayoutManager(layoutManager);
+            trashBinRecyclerView.setAdapter(trashBinAdapter);
+        }
+        setUpUi();
+    }
+
+    private ArrayList<TrashBinRealmModel> getTrashObjects(){
+        ArrayList<TrashBinRealmModel> list = new ArrayList<>();
+        for(int i = 0; i < trashBinRealmModelRealmQuery.count(); i++){
+            list.add(trashBinRealmModelRealmQuery.findAll().get(i));
+        }
+        return list;
+    }
+
+    private int columnsCount() {
+        return getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT
+                ? 2
+                : 3;
+    }
+
+    private void setUpUi(){
+        parentView.setBackgroundColor(getBackgroundColor());
+        setupToolbar();
+        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override public void onRefresh() {
+                trashBinAdapter.setResults(getTrashObjects());
+                if(swipeRefreshLayout.isRefreshing()){
+                    swipeRefreshLayout.setRefreshing(false);
+                }
+            }
+        });
+    }
+
+    private void setupToolbar(){
+        setSupportActionBar(toolbar);
+        //toolbar.setTitle("Trash Bin");
+        getSupportActionBar().setTitle("Trash Bin");
+        toolbar.setPopupTheme(getPopupToolbarStyle());
+        toolbar.setBackgroundColor(getPrimaryColor());
+        toolbar.setNavigationIcon(
+                new IconicsDrawable(this)
+                        .icon(CommunityMaterial.Icon.cmd_arrow_left)
+                        .color(Color.WHITE)
+                        .sizeDp(19));
+        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
```

Comment:
If we are using Java 8 ART then use lambda expressions instead of anonymous classes.

Reply:
@srv-twry I guess it's not necessary to use lambda expressions here, it's just fine as it is.